### DISCUSSION TITLE
Is there a way to mix email/password and passwordless sign in?

### DISCUSSION THREAD

**Discussion by ecomba** [mikker/passwordless#183] [Q&A] [answered]
We are thinking of allowing our users to sign in with email/password as well as passswordless (currently the sign in only works with passwordless) and I was wondering if this is possible using this gem?

**Comment by mikker** [comment, marked as answer]
It is definitely possible but you will have to build a custom `SessionsController` and handle the passwords yourself and all that. You will build most things yourself but you can definitely use Passwordless' built-in session handling and `ControllerHelpers`.
